internal/services/applications/application_password_resource.go

```diff
@@ -201,7 +202,17 @@ func applicationPasswordResourceRead(ctx context.Context, d *schema.ResourceData
 	}
 
 	tf.Set(d, "application_object_id", id.ObjectId)
-	tf.Set(d, "display_name", credential.DisplayName)
+
+	if credential.DisplayName != nil {
+		tf.Set(d, "display_name", credential.DisplayName)
+	} else {
```

Comment:
This will cause a crash if both `displayName` and `customKeyIdentifier` are null - which is the case for new password resources where `display_name` is omitted

```suggestion
	} else if credential.CustomKeyIdentifier != nil {
```